proposals/lip-define_new_transaction_schema.md

```diff
@@ -0,0 +1,241 @@
+
+
+## Rationale
+
+
+### Simplifying identification
+
+The type identifiers in the Lisk protocol are of type `uint32` or of type `bytes`. In both cases, they are set to a non-intuitive value which has to be memorized by users and developers. As the ecosystem grows and more functionalities are added, especially with the intoduction of interoperability, the number of values that have to be memorized gets quite large, making the identification system quite impractical (and potentially error-prone). On the other hand, for each component (modules, commands, events, etc.) there is a much more intuitive parameter: its name. Switching identifiers to those names (and removing the old identifiers) makes the whole user and developer experience easier and less error-prone. 
+
+Therefore, for the cases where removing the old identifiers does not introduce any significant challenge in the ecosystem, it is plausible to proceed to this switch and use names as identifiers. This is the case for modules, commands and events.  
+
+
+### New Property Names
+
+
+All properties in the proposed transaction schema are equivalent to the ones defined in [LIP 0028][lip-0028]. The only changes are the replace of identifiers by the corresponding names and the update of terminology according to the [LIP "Update Lisk SDK modular blockchain architecture"][lip-update-lisk-sdk-modular-architecture] (renaming a module asset to a command and a transaction `asset` property to `params`). Overall, `moduleID` is replaced by `module`, `assetID` is replaced by `command` and `asset` is renamed to `params`.
+
+
+## Specification
+
+The transaction schema defined in [LIP 0028][lip-0028] is superseded by the one defined [below](#json-schema). 
+
+The `params` property must follow the schema corresponding to the (`module`, `command`) pair defined in the corresponding module; we call this schema `paramsSchema`.
+
+As for the other transaction procedures:
+
+- Serialization and deserialization follow the same specifications already defined in [LIP 0028][lip-0028]; for completeness we include the pseudocode [below](#serialization). The resulting serialization is however different when the proposed transaction schema is used, due to the change of types for identifiers for module and command. Moreover, the transaction ID is calculated in the same way as described in [LIP 0028][lip-0028] (the SHA-256 hash of the serialized transaction object).
+- Signature calculation follows the same specifications as in [LIP 0028][lip-0028], updated to incorporate message tags introduced in [LIP 0037][lip-0037]. For completeness we provide the pseudocode [below](#transaction-signature-calculation).
+- Signature validation is done using the `verifySignatures` function defined in [LIP 0041](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0041.md#transaction-verification). 
```

Comment:
I would suggest to add a stateless function 
```
validateTransactionSignature(pk: PublicKeyEd25519, sig: SignatureEd25519, networkIdentifier, unsignedSerializedTransaction: bytes) -> bool:
```
This way we define the transaction signing and signature validation fully in one place (i.e., concrete call to the crypto library and which message tag is used) and everywhere else could utilize the functions here.

Reply:
I'm not sure if I agree that we need this stateless function. Concretely, the full transaction signature validation is delegated to the Auth module, and it is not stateless. Of course we could add this stateless version here, but why should it be attached in particular to the concept of transaction? the stateless signature verification is related more to validating a signature of a raw bytes object (and it is already defined in https://github.com/LiskHQ/lips/blob/main/proposals/lip-0062.md#specification)

Reply:
I have added a bulletpoint in the specification that signature validation is done using the `verifySignatures` function defined in LIP0041. I think this is enough. Let me know if you think otherwise. 

I could move it from a bulletpoint in the specification to a separate section after "Transaction Signature Calculation"

Reply:
Okay, that is also a good solution. Can be resolved from my side.